app/Http/Controllers/Cooperation/Admin/RoleController.php

```diff
@@ -26,31 +29,33 @@ public function assignRole(Cooperation $cooperation, Request $request)
         } else {
             $user = User::find($userId);
         }
-        $this->authorize('assign-role', $role);
 
-        abort_if($user->id === Hoomdossier::user()->id, 403);
+        $this->authorize('view', [$role, Hoomdossier::user(), \App\Helpers\HoomdossierSession::getRole(true)]);
+
 
         $user->assignRole($role);
 
         return redirect()->back();
     }
 
-    public function removeRole(Cooperation $cooperation, Request $request)
+    public function removeRole(UserRoleService $userRoleService, Cooperation $cooperation, Request $request)
     {
+        $currentUser = Hoomdossier::user();
         // the user id to assign the role to
         $userId = $request->get('user_id');
-        // the role we want to assign to a user
+        // the role we want to remove from the a user
         $roleId = $request->get('role_id');
-
         $role = Role::findById($roleId);
 
         // if the user is a super-admin, then hey may be removing roles from a user from a other cooperation, so we remove the scope.
-        if (Hoomdossier::user()->hasRoleAndIsCurrentRole('super-admin')) {
+        if ($currentUser->hasRoleAndIsCurrentRole('super-admin')) {
             $user = User::withoutGlobalScopes()->find($userId);
```

Comment:
Isn't it better to remove the dedicated scope (in this case for the cooperation)?

Reply:
yes!